test/test_ion_writer.cpp

```diff
@@ -32,7 +32,7 @@ class WriterTest : public ::testing::Test {
     FILE *out;
 };
 
-iERR ion_test_open_file_writer(hWRITER *writer, FILE *out, BOOL is_binary) {
```

Comment:
`ion_test_open_file_writer` creates a stream that is not freed when the writer is closed. The caller is not given the stream so knowledge of the stream, and how to free it using the writer is needed. I updated the function to return the stream as an argument so that callers will be aware and can easily free it.